README.md

```diff
@@ -6,22 +6,69 @@ Independent Parallel Particle Layer (IPPL) is a performance portable C++ library
 
-## Installing IPPL and its dependencies
+# Installing IPPL and its dependencies
 
 All the new developments of IPPL are merged into the `master` branch which can make it potentially unstable from time to time. So if you want a stable and more tested version
 please checkout the tagged branch correspodning to the latest release (e.g. `git checkout tags/IPPL-x.x.x`). Otherwise if you want the latest developments go with the master with the above caveat in mind.
 
-### Requirements
-The following libraries are required:
+## Requirements
 
+* [CMake](https://cmake.org/download/)
+* A C++ compilation toolchain (GPU-capable for GPU builds, e.g. [nvcc](https://developer.nvidia.com/cuda-downloads), [clang]() or [rocmcc](https://rocm.docs.amd.com/en/docs-5.0.2/reference/rocmcc/rocmcc.html))
 * MPI (GPU-aware if building for GPUs)
-* [Kokkos](https://github.com/kokkos) >= 4.1.00
-* [HeFFTe](https://github.com/icl-utk-edu/heffte) >= 2.2.0; only required if IPPL is built with FFTs enabled (`ENABLE_FFT=ON`)
 
-To build IPPL and its dependencies, we recommend using the [IPPL build scripts](https://github.com/IPPL-framework/ippl-build-scripts). See the [documentation](https://github.com/IPPL-framework/ippl-build-scripts#readme) for more info on how to use the IPPL build script.
+### Optional requirements
+* FFTW
+* CuFFT
+
+## Compilation
+IPPL is a CMake Project and can be configured by passing options in CMake syntax:
+```
+cmake <src_dir> -D<option>=<value>
+```
+#### None of the options have to be set explicitly, all have a default.
+
+The relevant options of IPPL are
+- IPPL_PLATFORMS, can be one of `SERIAL, OPENMP, CUDA, "OPENMP;CUDA"`, default `SERIAL`
+- `KOKKOS_VERSION`, default `4.1.00`
+- `HEFFTE_VERSION`, default `MASTER`
+- `ENABLE_SOLVERS`, default `OFF`
+- `ENABLE_FFT`, default `OFF`
+  - If `ENABLE_FFT` is set, `Heffte_ENABLE_CUDA` will default to `ON` if `IPPL_PLATFORMS` contains `cuda`
+  - Otherwise, `Heffte_ENABLE_AVX2` is enabled. FFTW has to be enabled explicitly.
+- `Heffte_ENABLE_FFTW`, default `OFF` 
+- `ENABLE_TESTS`, default `OFF`
+- `ENABLE_UNIT_TESTS`, default `OFF`
+- `ENABLE_ALPINE`, default `OFF`
+
+Furthermore, be aware of `CMAKE_BUILD_TYPE`, which can be either
+- `Release` for optimized builds
+- `RelWithDebInfo` for optimized builds with debug info (default)
+- `Debug` for debug builds (with [**Sanitizers enabled**](https://gcc.gnu.org/onlinedocs/gcc-13.2.0/gcc/Instrumentation-Options.html))
+
+### Examples
+Download and setup a build directory:
+```
+https://github.com/IPPL-framework/ippl
+cd ippl
+mkdir build
+cd build
+```
+#### Serial debug build with tests and newest Kokkos
+```
+cmake .. -DCMAKE_BUILD_TYPE=Debug -DENABLE_TESTS=True -DKOKKOS_VERSION=4.2.00
+```
+#### OpenMP release build with tests and FFTW
+```
+cmake .. -DCMAKE_BUILD_TYPE=Release -DIPPL_PLATFORMS=openmp -DHeffte_ENABLE_FFTW -DENABLE_TESTS=True
```

Comment:
Don't you have to point to FFTW installation directory through heffte options in this case?

Reply:
I think my cmake and the 200-build-heffte script both rely on Heffte to find fftw, and it works for me locally and on merlin.

Reply:
Yeah but in the build scripts we do specify this https://github.com/IPPL-framework/ippl-build-scripts/blob/main/200-build-heffte#L60C5-L60C30 . Are you also specifying it somewhere?

Reply:
You mean `FFTW_DIR`? If you install FFTW as a module commonly that environment variable is also set. Otherwise you have to manually set it before executing the script.

Reply:
Ah yes, then it's picked up by cmake as well.

Reply:
So are you saying we don't need to explicitly specify this `-DFFTW_ROOT="${FFTW_DIR}"` cmake option in this line and cmake automatically picks it up?

Reply:
I just corrected this example from `-DHeffte_ENABLE_FFTW` to `-DHeffte_ENABLE_FFTW=True`, and all I can say with certainty that `cmake .. -DCMAKE_BUILD_TYPE=Release -DIPPL_PLATFORMS=openmp -DHeffte_ENABLE_FFTW=True -DENABLE_TESTS=True -DENABLE_FFT=True -DENABLE_SOLVERS=True` and ` TestFFTPeriodicPoissonSolver` compiles.